doc/how-to/contribute.rst

```diff
@@ -0,0 +1,150 @@
+
+Pull requests
+-------------
+
+Propose your changes to this project as pull requests on `GitHub`_.
+
+Proposed changes will then go through review there and once approved, be merged in the main branch.
+
+Commit structure
+~~~~~~~~~~~~~~~~
+
+Use separate commits for every logical change, and for changes to different components.
+Prefix your commits with the component that is affected, using the code tree structure.
+For example, prefix a commit that updates the MicroCloud service with ``microcloud/service:``.
+
+This structure makes it easier for contributions to be reviewed and also greatly simplifies the process of porting fixes to other branches.
+
+Developer Certificate of Origin
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To improve tracking of contributions to this project we use the DCO 1.1 and use a “sign-off” procedure for all changes going into the branch.
+
+The sign-off is a simple line at the end of the explanation for the commit which certifies that you wrote it or otherwise have the right to pass it on as an open-source contribution.
+
+.. code::
+
+   Developer Certificate of Origin
+   Version 1.1
+
+   Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
+   660 York Street, Suite 102,
+   San Francisco, CA 94110 USA
+
+   Everyone is permitted to copy and distribute verbatim copies of this
+   license document, but changing it is not allowed.
+
+   Developer's Certificate of Origin 1.1
+
+   By making a contribution to this project, I certify that:
+
+   (a) The contribution was created in whole or in part by me and I
+       have the right to submit it under the open source license
+       indicated in the file; or
+
+   (b) The contribution is based upon previous work that, to the best
+       of my knowledge, is covered under an appropriate open source
+       license and I have the right under that license to submit that
+       work with modifications, whether created in whole or in part
+       by me, under the same open source license (unless I am
+       permitted to submit under a different license), as indicated
+       in the file; or
+
+   (c) The contribution was provided directly to me by some other
+       person who certified (a), (b) or (c) and I have not modified
+       it.
+
+   (d) I understand and agree that this project and the contribution
+       are public and that a record of the contribution (including all
+       personal information I submit with it, including my sign-off) is
+       maintained indefinitely and may be redistributed consistent with
+       this project or the open source license(s) involved.
+
+An example of a valid sign-off line is::
+
+  Signed-off-by: Random J Developer <random@developer.org>
+
+Use a known identity and a valid e-mail address.
+Sorry, no anonymous contributions are allowed.
+
+We also require each commit be individually signed-off by their author, even when part of a larger set.
+You may find git :command:`commit -s` useful.
```

Comment:
This is partly rewritten and partly based on LXD. Is it what we want?

Reply:
Yes this seems good. I think we also need to mention that signing the CLA is required for this repo.

Reply:
I agree we should clearly indicate it.